go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go

```diff
@@ -0,0 +1,491 @@
+	}
+
+	return nil
+}
+
+func clearTableAndUpdateRoot(ctx *sql.Context, root *doltdb.RootValue, tbl *doltdb.Table, tblName string) (*doltdb.RootValue, error) {
+	newTbl, err := tbl.ClearConflicts(ctx)
+	if err != nil {
+		return nil, err
+	}
+	newRoot, err := root.PutTable(ctx, tblName, newTbl)
+	if err != nil {
+		return nil, err
+	}
+	return newRoot, nil
+}
+
+func ResolveConflicts(ctx *sql.Context, dEnv *env.DoltEnv, dSess *dsess.DoltSession, root *doltdb.RootValue, dbName string, ours bool, tblNames []string) error {
+	for _, tblName := range tblNames {
+		tbl, ok, err := root.GetTable(ctx, tblName)
+		if err != nil {
+			return err
+		}
+		if !ok {
+			return doltdb.ErrTableNotFound
+		}
+
+		if has, err := tbl.HasConflicts(ctx); err != nil {
+			return err
+		} else if !has {
+			return nil
+		}
+
+		sch, err := tbl.GetSchema(ctx)
+		if err != nil {
+			return err
+		}
+		_, ourSch, theirSch, err := tbl.GetConflictSchemas(ctx, tblName)
+		if err != nil {
+			return err
+		}
+
+		if ours && !schema.ColCollsAreEqual(sch.GetAllCols(), ourSch.GetAllCols()) {
+			return ErrConfSchIncompatible
+		} else if !ours && !schema.ColCollsAreEqual(sch.GetAllCols(), theirSch.GetAllCols()) {
+			return ErrConfSchIncompatible
+		}
+
+		if !ours {
+			if tbl.Format() == types.Format_DOLT {
+				tbl, err = resolveProllyConflicts(ctx, tbl, tblName, sch)
+			} else {
+				tbl, err = resolveNomsConflicts(ctx, dEnv, tbl, tblName, sch)
+			}
+			if err != nil {
+				return err
+			}
+		}
+
+		newRoot, err := clearTableAndUpdateRoot(ctx, root, tbl, tblName)
+		if err != nil {
+			return err
+		}
+
+		err = validateConstraintViolations(ctx, root, newRoot, tblName)
+		if err != nil {
+			return err
+		}
+
+		root = newRoot
+	}
+	return dSess.SetRoot(ctx, dbName, root)
+}
+
+func DoDoltConflictsResolve(ctx *sql.Context, args []string) (int, error) {
+	dbName := ctx.GetCurrentDatabase()
+	fmt.Printf("database name: %s", dbName)
+
+	apr, err := cli.CreateConflictsResolveArgParser().Parse(args)
+	if err != nil {
+		return 1, err
+	}
+
+	dSess := dsess.DSessFromSess(ctx.Session)
+	roots, ok := dSess.GetRoots(ctx, dbName)
+	if !ok {
+		return 1, fmt.Errorf("Could not load database %s", dbName)
+	}
+
+	ours := apr.Contains(cli.OursFlag)
+	theirs := apr.Contains(cli.TheirsFlag)
+	if ours && theirs {
+		return 1, fmt.Errorf("specify only either --ours or --theirs")
+	} else if !ours && !theirs {
+		return 1, fmt.Errorf("--ours or --theirs must be supplied")
+	}
+
+	if apr.NArg() == 0 {
+		return 1, fmt.Errorf("specify at least one table to resolve conflicts")
+	}
+
+	// get all tables in conflict
+	root := roots.Working
+	tbls := apr.Args
+	if len(tbls) == 1 && tbls[0] == "." {
+		if allTables, err := root.TablesInConflict(ctx); err != nil {
+			return 1, err
+		} else {
+			tbls = allTables
+		}
+	}
+
+	dEnv := env.Load(ctx, env.GetCurrentUserHomeDir, filesys.LocalFS, doltdb.LocalDirDoltDB, "")
```

Comment:
the dolt environment should only be created at startup. get the editor options from the session:

```diff
diff --git a/go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go b/go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go
index 0dea4b326..743c26816 100644
--- a/go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go
+++ b/go/libraries/doltcore/sqle/dfunctions/dolt_conflicts_resolve.go
@@ -26,14 +26,12 @@ import (
 	"github.com/dolthub/dolt/go/libraries/doltcore/conflict"
 	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb"
 	"github.com/dolthub/dolt/go/libraries/doltcore/doltdb/durable"
-	"github.com/dolthub/dolt/go/libraries/doltcore/env"
 	"github.com/dolthub/dolt/go/libraries/doltcore/merge"
 	"github.com/dolthub/dolt/go/libraries/doltcore/row"
 	"github.com/dolthub/dolt/go/libraries/doltcore/schema"
 	"github.com/dolthub/dolt/go/libraries/doltcore/sqle/dsess"
 	"github.com/dolthub/dolt/go/libraries/doltcore/table"
 	"github.com/dolthub/dolt/go/libraries/doltcore/table/editor"
-	"github.com/dolthub/dolt/go/libraries/utils/filesys"
 	"github.com/dolthub/dolt/go/libraries/utils/set"
 	"github.com/dolthub/dolt/go/store/hash"
 	"github.com/dolthub/dolt/go/store/prolly"
@@ -205,13 +203,8 @@ func resolveProllyConflicts(ctx *sql.Context, tbl *doltdb.Table, tblName string,
 	return newTbl, nil
 }
 
-func resolvePkConflicts(ctx *sql.Context, dEnv *env.DoltEnv, tbl *doltdb.Table, tblName string, sch schema.Schema, conflicts types.Map) (*doltdb.Table, error) {
+func resolvePkConflicts(ctx *sql.Context, opts editor.Options, tbl *doltdb.Table, tblName string, sch schema.Schema, conflicts types.Map) (*doltdb.Table, error) {
 	// Create table editor
-	tmpDir, err := dEnv.TempTableFilesDir()
-	if err != nil {
-		return nil, err
-	}
-	opts := editor.Options{Deaf: dEnv.DbEaFactory(), Tempdir: tmpDir}
 	tblEditor, err := editor.NewTableEditor(ctx, tbl, sch, tblName, opts)
 	if err != nil {
 		return nil, err
@@ -307,7 +300,7 @@ func resolveKeylessConflicts(ctx *sql.Context, tbl *doltdb.Table, conflicts type
 	return tbl.UpdateNomsRows(ctx, rowData)
 }
 
-func resolveNomsConflicts(ctx *sql.Context, dEnv *env.DoltEnv, tbl *doltdb.Table, tblName string, sch schema.Schema) (*doltdb.Table, error) {
+func resolveNomsConflicts(ctx *sql.Context, opts editor.Options, tbl *doltdb.Table, tblName string, sch schema.Schema) (*doltdb.Table, error) {
 	// Get conflicts
 	_, confIdx, err := tbl.GetConflicts(ctx)
 	if err != nil {
@@ -319,7 +312,7 @@ func resolveNomsConflicts(ctx *sql.Context, dEnv *env.DoltEnv, tbl *doltdb.Table
 		return resolveKeylessConflicts(ctx, tbl, conflicts)
 	}
 
-	return resolvePkConflicts(ctx, dEnv, tbl, tblName, sch, conflicts)
+	return resolvePkConflicts(ctx, opts, tbl, tblName, sch, conflicts)
 }
 
 func validateConstraintViolations(ctx *sql.Context, before, after *doltdb.RootValue, table string) error {
@@ -352,7 +345,7 @@ func clearTableAndUpdateRoot(ctx *sql.Context, root *doltdb.RootValue, tbl *dolt
 	return newRoot, nil
 }
 
-func ResolveConflicts(ctx *sql.Context, dEnv *env.DoltEnv, dSess *dsess.DoltSession, root *doltdb.RootValue, dbName string, ours bool, tblNames []string) error {
+func ResolveConflicts(ctx *sql.Context, dSess *dsess.DoltSession, root *doltdb.RootValue, dbName string, ours bool, tblNames []string) error {
 	for _, tblName := range tblNames {
 		tbl, ok, err := root.GetTable(ctx, tblName)
 		if err != nil {
@@ -383,11 +376,17 @@ func ResolveConflicts(ctx *sql.Context, dEnv *env.DoltEnv, dSess *dsess.DoltSess
 			return ErrConfSchIncompatible
 		}
 
+		state, _, err := dSess.LookupDbState(ctx, dbName)
+		if err != nil {
+			return err
+		}
+		opts := state.WriteSession.GetOptions()
+
 		if !ours {
 			if tbl.Format() == types.Format_DOLT {
 				tbl, err = resolveProllyConflicts(ctx, tbl, tblName, sch)
 			} else {
-				tbl, err = resolveNomsConflicts(ctx, dEnv, tbl, tblName, sch)
+				tbl, err = resolveNomsConflicts(ctx, opts, tbl, tblName, sch)
 			}
 			if err != nil {
 				return err
@@ -447,8 +446,7 @@ func DoDoltConflictsResolve(ctx *sql.Context, args []string) (int, error) {
 		}
 	}
 
-	dEnv := env.Load(ctx, env.GetCurrentUserHomeDir, filesys.LocalFS, doltdb.LocalDirDoltDB, "")
-	err = ResolveConflicts(ctx, dEnv, dSess, root, dbName, ours, tbls)
+	err = ResolveConflicts(ctx, dSess, root, dbName, ours, tbls)
 	if err != nil {
 		return 1, err
 	}
```